### PR TITLE
sentinel: Vulnerability database syncer routine

### PR DESCRIPTION
**Pull request by efritz**
This PR implements the background routine that will periodically check for new vulnerabilities from the GitHub advisory database and upsert them into Postgres.

## Test plan

Tested locally.

🥚

### CODE REVIEW

File: enterprise/internal/codeintel/sentinel/config.go [hunk 1]
```diff
@@ -9,11 +9,13 @@ import (
...
+	DownloaderInterval time.Duration
+	MatcherInterval    time.Duration
 }
 
 var ConfigInst = &sentinelConfig{}
 
 func (c *sentinelConfig) Load() {
-	c.Interval = c.GetInterval("CODEINTEL_SENTINEL_INTERVAL", "1s", "How frequently to run background jobs.")
+	c.DownloaderInterval = c.GetInterval("CODEINTEL_SENTINEL_DOWNLOADER_INTERVAL", "1h", "How frequently to sync the vulnerability database.")
+	c.MatcherInterval = c.GetInterval("CODEINTEL_SENTINEL_MATCHER_INTERVAL", "1s", "How frequently to match existing records against known vulnerabilities.")
```
Comment (eseliger) [inline enterprise/internal/codeintel/sentinel/config.go:20]: can we start moving some of these in site config instead? we seem to be hoarding _loads_ of env vars again

File: enterprise/internal/codeintel/sentinel/internal/background/cve_source_github.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+	"path/filepath"
+	"time"
+
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/sourcegraph/sourcegraph/enterprise/internal/codeintel/sentinel/shared"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+const advisoryDatabaseURL = "https://github.com/github/advisory-database/archive/refs/heads/main.zip"
```
Comment (eseliger) [inline enterprise/internal/codeintel/sentinel/internal/background/cve_source_github.go:23]: should we host a mirror of this file? GitHub kinda dislikes lots of file fetches and might rate limit

File: enterprise/internal/codeintel/sentinel/internal/background/cve_source_github.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+// ReadGitHubAdvisoryDB fetches a copy of the GHSA database and converts it to the internal Vulnerability format
+func (parser *CVEParser) ReadGitHubAdvisoryDB(ctx context.Context, useLocalCache bool) (vulns []shared.Vulnerability, err error) {
+	if useLocalCache {
+		zipReader, err := os.Open("main.zip")
```
Comment (eseliger) [inline enterprise/internal/codeintel/sentinel/internal/background/cve_source_github.go:28]: no cache dir?

**Review by eseliger — APPROVED** [review]
solid stuff!

File: enterprise/internal/codeintel/sentinel/config.go (hunk 1 above)
Comment (efritz) [reply enterprise/internal/codeintel/sentinel/config.go:20]: Yes, but sentinel is dotcom only for now, so I'd like to advertise this as little as possible outside of the necessary code to tweak things.

I agree that most of these should become site config vars, but I'd like to make that switch once we know what the system actually looks like apart from a demo page.

File: enterprise/internal/codeintel/sentinel/internal/background/cve_source_github.go (hunk 3 above)
Comment (efritz) [reply enterprise/internal/codeintel/sentinel/internal/background/cve_source_github.go:28]: I think this was here just for @willdollman to test; have left all of the `cve_souuce_*.go` files mainly alone. Will extract these post-merge into a lib/shared package as well.

**Pull request merged**